### PR TITLE
Exclude full OAuth token details from printing out on stdout

### LINKED ISSUE #55425: OAuth token is displayed in plain text in debug log
**What happened**:
When debug log is enabled, the OAuth token response is printed completely without any obfuscation

**What you expected to happen**:
No need to print the full json response from OAuth provider

**How to reproduce it (as minimally and precisely as possible)**:
Enable debug logs and login via OAuth provider.

**Anything else we need to know?**:

**Environment**:
- Grafana version: `v8.3.3 (30bb7a93ca)`
- Data source type & version:
- OS Grafana is installed on:
- User OS & Browser:
- Grafana plugins:
- Others:

### PR DESCRIPTION
**Pull request by junaid-ali**
**What this PR does / why we need it**:
When debug log is enabled, the OAuth token response is displayed completely in plain text

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana/issues/55425

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by Jguer** [comment]
Hi @junaid-ali , these logs are only printed if the instance is running in debug level, which should not be used in production, only for debugging. 

Printing the full json response from OAuth provider is useful in a development and debugging environment. Is there any reason that forces you to run debug mode in a production instance?

**Comment by junaid-ali** [comment]
@Jguer I was running it in development to debug some issues with the OAuth provider, but IMO the full json response with those credentials in plain text shouldn't be needed. We can print other non-sensitive fields which can still validate that we have received a valid response. If someone unknowingly runs Grafana in debug mode in staging or production, the workload logs are usually exported to other log handlers, causing the credentials to be spread around.

**Comment by jmatosgrafana** [comment]
Security wise I believe it's a good choice not to log the token in debug mode (e.g. misconfig of log levels), as long as there are other ways to debug (e.g. setting up a proxy to view the token while debugging)

**Comment by junaid-ali** [comment]
@jmatosgrafana I was testing it with gitlab, which prints out the /user endpoint response in debug mode, which should be enough to validate the token (I assume other providers do print out such responses as well). My other concern is that we are printing the refresh token alongside the access token as well.

File: pkg/api/login_oauth.go [hunk 1]
```diff
@@ -194,7 +194,7 @@ func (hs *HTTPServer) OAuthLogin(ctx *models.ReqContext) {
 	// token.TokenType was defaulting to "bearer", which is out of spec, so we explicitly set to "Bearer"
 	token.TokenType = "Bearer"
 
-	oauthLogger.Debug("OAuthLogin: got token", "token", fmt.Sprintf("%+v", token))
+	oauthLogger.Debug("OAuthLogin: got token", " and expires at", fmt.Sprintf("%v", token.Expiry))
```
Comment (IevaVasiljeva) [inline pkg/api/login_oauth.go:197]:
```suggestion
	oauthLogger.Debug("OAuthLogin: got token", "expiry", fmt.Sprintf("%v", token.Expiry))
```

**Review by IevaVasiljeva — APPROVED** [review]
We've discussed this internally, and agree with you that it makes sense to remove the token from the logs and only log specific information contained in the token that would help with debugging.

We can extend this log line with more info from the token in the future, but this looks good for now.

File: pkg/api/login_oauth.go (hunk 1 above)
Comment (junaid-ali) [reply pkg/api/login_oauth.go]: done

**Comment by junaid-ali** [comment]
Hi @IevaVasiljeva, could you please have a look it again?

**Review by Jguer — APPROVED** [review]

**Comment by Jguer** [comment]
Sorry for the delay @junaid-ali , thanks for the patch, I'll merge it in as soon as CI passes

**Pull request merged**